test/integration/targets/ansible-galaxy-collection/tasks/upgrade.yml

```diff
@@ -0,0 +1,166 @@
+##### Updating collections with a new version constraint
+
+# No deps
+
+- name: reset installation directory
+  file:
+    state: "{{ item }}"
+    path: "{{ galaxy_dir }}/ansible_collections"
+  loop:
+    - absent
+    - directory
+
+- name: install a collection
+  command: ansible-galaxy collection install namespace1.name1:0.0.1 -s pulp_v3 {{ galaxy_verbosity }}
+  register: result
+  failed_when:
+    - '"namespace1.name1:0.0.1 was installed successfully" not in result.stdout_lines'
+
+- name: install a new version of the collection without --force
+  command: ansible-galaxy collection install namespace1.name1:>0.0.4,<=0.0.5 -s pulp_v3 {{ galaxy_verbosity }}
+  register: result
+
+- assert:
+    that: '"namespace1.name1:0.0.5 was installed successfully" in result.stdout_lines'
+
+- name: don't reinstall the collection in the requirement is compatible
+  command: ansible-galaxy collection install namespace1.name1:>=0.0.5 -s pulp_v3 {{ galaxy_verbosity }}
+  register: result
+
+- assert:
+    that: "\"Nothing to do. All requested collections are already installed.\" in result.stdout"
+
+- name: install a pre-release of the collection without --force
+  command: ansible-galaxy collection install namespace1.name1:1.1.0-beta.1 -s pulp_v3 {{ galaxy_verbosity }}
+  register: result
+
+- assert:
+    that: '"namespace1.name1:1.1.0-beta.1 was installed successfully" in result.stdout_lines'
+
+# With deps
+
+- name: reset installation directory
+  file:
+    state: "{{ item }}"
+    path: "{{ galaxy_dir }}/ansible_collections"
+  loop:
+    - absent
+    - directory
+
+- name: install a dep that will need to be upgraded to be compatible with the parent
+  command: ansible-galaxy collection install child_dep.child_collection:0.4.0 --no-deps -s pulp_v3 {{ galaxy_verbosity }}
+  register: result
```

Comment:
Is registering `result` necessary when it's unused?

Reply:
It is used in `failed_when`. I used `failed_when` to make sure setup tasks did what was expected and `assert` for things I'm testing.